lang/cs/Org.Apache.REEF.IMRU.Examples/Run.cs

```diff
@@ -32,8 +32,121 @@ namespace Org.Apache.REEF.IMRU.Examples
+            }
+
+            var mapperCountExample = injector.GetInstance<MapperCount.MapperCount>();
+            mapperCountExample.Run(numNodes - 1);
+        }
+
+        private static void RunBroadcastReduceTest(IConfiguration tcpPortConfig, bool runOnYarn, int numNodes, string[] args)
+        {
+            int chunkSize = 2;
+            int dims = 10;
+            int iterations = 10;
+
+            if (args.Length > 0)
+            {
+                dims = Convert.ToInt32(args[0]);
+            }
+
+            if (args.Length > 1)
+            {
+                chunkSize = Convert.ToInt32(args[1]);
+            }
+
+            if (args.Length > 2)
+            {
+                iterations = Convert.ToInt32(args[2]);
+            }
+
+            IInjector injector;
+
+            if (!runOnYarn)
+            {
+                injector =
+                    TangFactory.GetTang()
+                        .NewInjector(OnREEFIMRURunTimeConfiguration<int[], int[], int[]>.GetLocalIMRUConfiguration(numNodes), tcpPortConfig);
+            }
+            else
+            {
+                injector = TangFactory.GetTang()
+                    .NewInjector(OnREEFIMRURunTimeConfiguration<int, int, int>.GetYarnIMRUConfiguration(), tcpPortConfig);
+            }
+            var broadcastReduceExample = injector.GetInstance<PipelinedBroadcastAndReduce>();
+            broadcastReduceExample.Run(numNodes - 1, chunkSize, iterations, dims);
+        }
+
         private static void Main(string[] args)
-        {      
+        {
+            Console.WriteLine("start running client: " + DateTime.Now);
+            string methodName = "MapperCount";
+            bool runOnYarn = false;
+            int numNodes = 2;
+            int startPort = 8900;
+            int portRange = 1000;
+
+            if (args != null)
+            {
+                if (args.Length > 0)
+                {
+                    runOnYarn = bool.Parse(args[0].ToLower());
+                }
+
+                if (args.Length > 1)
+                {
+                    numNodes = int.Parse(args[1]);
+                }
+
+                if (args.Length > 2)
+                {
+                    startPort = int.Parse(args[2]);
+                }
+
+                if (args.Length > 3)
+                {
+                    portRange = int.Parse(args[3]);
+                }
+
+                if (args.Length > 4)
+                {
+                    methodName = args[4];
+                }
+            }
+
+            var tcpPortConfig = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindNamedParameter(typeof (TcpPortRangeStart),
+                    startPort.ToString(CultureInfo.InvariantCulture))
+                .BindNamedParameter(typeof (TcpPortRangeCount),
+                    portRange.ToString(CultureInfo.InvariantCulture))
+                .Build();
+
+            if (methodName.ToLower().Equals("mappercount"))
```

Comment:
Maybe a switch would be more appropriate? That way, you can die if no known method was given.